docs/modules/ROOT/pages/how-to-guides/Import-code/proc_importing_code.adoc

```diff
@@ -0,0 +1,69 @@
+= Importing code to App Studio
+
+[role="_abstract"]
+After you click **Create an application**, the **Grab some code** page opens. From here, choose one of the following source code options to begin building your application:
+
+* Import your own code.
+* Start with an {ProductName} code sample.
+
+== Importing code from your GitHub repository
+
+To use your own code to build your application, complete the following steps:
+
+. Enter the link to your repository in the **Git repository URL** field. {ProductName} verifies your URL right away. You don’t even have to press **Enter**.
+. Optional: Add a **Git reference** to point to code in a specific branch, tag, or commit that you want to use to build your application.
+. Optional: Indicate a **Context directory** to specify the subdirectory for the source code you want to use. 
+
+[NOTE]
+====
+Make sure this path is correct. You might get an error if your build context folder is your root directory but your Dockerfile is in a subdirectory of that folder.
+====
+[start=4]
+. Click **Import code**. 
+
+[TIP]
+====
+If {ProductName} can’t access your Git repository, see <<Granting App Studio access to your Git repository>> for help.
+====
+
+{ProductName} clones, scans, and analyzes your Git repository to determine how many components to add to your application and which runtime to use. The types of files in your repository, and also your Git directory file hierarchy, determine the number and type of components to add. For example, {ProductName} looks for devfiles and Dockerfiles, and also language-specific build configuration files like pom.xml for Java or package.json for Node.js.
+
+[NOTE]
+====    
+{ProductName} can build applications that are written in the following language stacks and frameworks:
+
+    * Python
+    * Java
+    * Go
+    * .NET
+    * Node.js
+    * Quarkus
+    * Spring Boot
```

Comment:
I'm not an expert here, but is it helpful to call out Java, Quarkus, and Spring Boot separately? You list 7 runtimes here, but we only offer 5 samples. There are two types of Java samples, Quarkus and Spring Boot, so there's no Java entry, and no .Net).

Reply:
thanks, Matt. a SME fed me this list but it's been weeks and i'd have to dig to remember who. should i simply delete the two you mention? rename the list to something that includes all 6? make a separate runtimes list? @mattreid

Reply:
My non-expert recommendation is consolidating Spring Boot and Quarkus into the Java line and dropping .Net for now, but @jduimovich is more qualified to give guidance here.